### PR TITLE
Rename *_with_libctx to *_ex

### PR DESCRIPTION
**Pull request by mattcaswell**
Rename all instances of _with_libctx to _ex as per the new coding style policy.

### CODE REVIEW

File: crypto/store/store_register.c [hunk 1]
```diff
@@ -70,11 +70,11 @@ int OSSL_STORE_LOADER_set_open(OSSL_STORE_LOADER *loader,
     return 1;
 }
 
-int OSSL_STORE_LOADER_set_open_with_libctx
+int OSSL_STORE_LOADER_set_open_ex
     (OSSL_STORE_LOADER *loader,
-     OSSL_STORE_open_with_libctx_fn open_with_libctx_function)
+     OSSL_STORE_open_ex_fn open_ex_function)
```
Comment (paulidale) [inline crypto/store/store_register.c:75]: I'm not sure I like _ex in a type name but can't think of anything better.

File: crypto/store/store_register.c [hunk 2]
```diff
@@ -222,7 +222,7 @@ const OSSL_STORE_LOADER *ossl_store_get0_loader_int(const char *scheme)
     template.load = NULL;
     template.eof = NULL;
     template.close = NULL;
-    template.open_with_libctx = NULL;
+    template.open_ex = NULL;
```
Comment (paulidale) [inline crypto/store/store_register.c:225]: Ditto for field names.

File: doc/man3/EVP_VerifyInit.pod [hunk 3]
```diff
@@ -49,7 +49,7 @@ implementation of digest I<type>.
 EVP_VerifyInit_ex() and EVP_VerifyUpdate() return 1 for success and 0 for
 failure.
 
-EVP_VerifyFinal()_with_libctx() and EVP_VerifyFinal() return 1 for a correct
+EVP_VerifyFinal()_ex() and EVP_VerifyFinal() return 1 for a correct
```
Comment (paulidale) [inline doc/man3/EVP_VerifyInit.pod:52]: The first `()` should be dropped -- existing error but might as well get it fixed.

**Review by paulidale — APPROVED** [review]
One (existing) typo in the doc which would be nice to fix.

File: doc/man3/EVP_VerifyInit.pod (hunk 3 above)
Comment (mattcaswell) [reply doc/man3/EVP_VerifyInit.pod]: Typo fixed

**Review by slontis — COMMENTED** [review]
I cant see any issues..

**Comment by slontis** [comment]
I wonder if there are going to be any _ex2 cases?

**Comment by t8m** [comment]
Should there be also parameter reordering in place according to the @levitte proposal? I mean putting the libctx as the first argument and the propq before any callback arguments if there are any? But I suppose that is for different PR as that won't be easily automated.

**Comment by t8m** [comment]
CIFuzz failure is relevant:
 In function `sm2_gen':
/src/openssl/providers/implementations/keymgmt/ec_kmgmt.c:1097: undefined reference to `EC_KEY_new_with_libctx'

**Comment by levitte** [comment]
> Should there be also parameter reordering in place according to the @levitte proposal? I mean putting the libctx as the first argument and the propq before any callback arguments if there are any? But I suppose that is for different PR as that won't be easily automated.

I'm doing some work in reduce the amount of _with_libctx-to-become-_ex functions (I have it my mind that they've been sprinkled wider than actually necessary, especially when keeping in mind that we want to make the transition easier on the user), and am rearranging the arguments to be a bit more consistent across the board.

**Comment by slontis** [comment]
Looks like it needs a rebase

**Comment by mattcaswell** [comment]
The CI build failures were because master had been updated to include a new reference to a `_with_libctx` function - so when the CIs merged this branch with master, the merge worked but the build failed. I suspect this might be an ongoing problem until this PR is merged.

I've updated it with a new fixup to address the problem. Please reconfirm!

**Comment by slontis** [comment]
@paulidale - LGTM - please reconfirm.

**Review by paulidale — APPROVED** [review]
Reconfirm

**Comment by mattcaswell** [comment]
Pushed.  I had to resolve a trivial conflict.

**Pull request closed without merging**